rln/src/ffi/ffi_utils.rs

```diff
@@ -162,25 +165,31 @@ pub fn bytes_be_to_vec_cfr(
 }
 
 #[ffi_export]
-pub fn vec_cfr_debug(v: &repr_c::Vec<CFr>) -> repr_c::String {
-    format!("{:?}", v.iter().map(|cfr| cfr.0).collect::<Vec<Fr>>()).into()
+pub fn vec_cfr_debug(v: Option<&repr_c::Vec<CFr>>) -> repr_c::String {
+    match v {
+        Some(v) => {
+            let vec_fr: Vec<Fr> = v.iter().map(|cfr| cfr.0).collect();
+            format!("{:?}", vec_fr).into()
+        }
+        None => "None".into(),
+    }
 }
 
 #[ffi_export]
-pub fn vec_cfr_free(v: repr_c::Vec<CFr>) {
+pub fn vec_cfr_free(v: repr_c::Box<repr_c::Vec<CFr>>) {
     drop(v);
 }
 
 // Vec<u8>
 
 #[ffi_export]
-pub fn vec_u8_to_bytes_le(vec: &repr_c::Vec<u8>) -> repr_c::Vec<u8> {
-    crate::utils::vec_u8_to_bytes_le(vec).into()
+pub fn vec_u8_to_bytes_le(vec: &repr_c::Vec<u8>) -> repr_c::Box<repr_c::Vec<u8>> {
+    Box_::new(crate::utils::vec_u8_to_bytes_le(vec).into())
 }
 
 #[ffi_export]
-pub fn vec_u8_to_bytes_be(vec: &repr_c::Vec<u8>) -> repr_c::Vec<u8> {
-    crate::utils::vec_u8_to_bytes_be(vec).into()
+pub fn vec_u8_to_bytes_be(vec: &repr_c::Vec<u8>) -> repr_c::Box<repr_c::Vec<u8>> {
+    Box_::new(crate::utils::vec_u8_to_bytes_be(vec).into())
```

Comment:
Unclear what is done here (need docstring?). a Vec<u8> is turned into another Vec<u8>?